images/egress/router/egress-router.sh

```diff
@@ -43,42 +46,61 @@ function setup_network() {
       arping -q -U -c 1 -I macvlan0 "${EGRESS_SOURCE}" || true ) > /dev/null 2>&1 < /dev/null &
 }
 
+function validate_port() {
+    local port=$1
+    if [[ "${port}" -lt "1" || "${port}" -gt "65535" ]]; then
+        die "Invalid port: ${port}, must be in the range 1 to 65535"
+    fi
+}
+
 function gen_iptables_rules() {
     if [[ -z "${EGRESS_DESTINATION:-}" ]]; then
-        echo "EGRESS_DESTINATION unspecified"
-        exit 1
+        die "EGRESS_DESTINATION unspecified"
     fi
 
     did_fallback=
+    declare -A used_ports=()
     while read dest; do
 	if [[ "${dest}" =~ ^${BLANK_LINE_OR_COMMENT_REGEX}$ ]]; then
             # comment or blank line
             continue
 	fi
 	if [[ -n "${did_fallback}" ]]; then
-            echo "EGRESS_DESTINATION fallback IP must be the last line" 1>&2
-            exit 1
+            die "EGRESS_DESTINATION fallback IP must be the last line" 1>&2
 	fi
 
+	localport_set=
 	if [[ "${dest}" =~ ^${IP_REGEX}$ ]]; then
             # single IP address: do fallback "all ports to same IP"
             echo -A PREROUTING -i eth0 -j DNAT --to-destination "${dest}"
             did_fallback=1
 
 	elif [[ "${dest}" =~ ^${PORT_REGEX}\ +${PROTO_REGEX}\ +${IP_REGEX}$ ]]; then
             read localport proto destip <<< "${dest}"
+	    localport_set=1
```

Comment:
meh... can see here that there's tabs-vs-spaces inconsistency in this file. maybe add another commit fully untabifying?